tools/prefetch_cache_gcsfuse/prefetch.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/googlecloudplatform/gcsfuse/internal/gcsx"
+	"google.golang.org/api/iterator"
+)
+
+func downloadFile(ctx context.Context, client *storage.Client, object *storage.ObjectAttrs, cacheDir string) (err error) {
+	log.Printf(fmt.Sprintf("downloading file %v from bucket %v into dir %v", object.Name, object.Bucket, cacheDir))
+
+	// TODO ezl store the file name inside the json metadata instead of stripping the metadata filename
+	tempFileObjectMetadata := &gcsx.TempFileObjectMetadata{
+		BucketName:     object.Bucket,
+		ObjectName:     object.Name,
+		Generation:     object.Generation,
+		MetaGeneration: object.Metageneration,
+	}
+
+	// TODO ezl refactor once we decouple content cache from gcsx
+	// Create a temporary cache file on disk
+	// TODO ezl concerns about partial download and cache file corruption
+	// We may want a way to verify the files are fully downloaded
+	// and either resuming the download or discarding and redownloading the file
+	f, err := ioutil.TempFile(cacheDir, gcsx.CACHE_FILE_PREFIX)
+
+	if err != nil {
+		err = fmt.Errorf("ioutil.TempFile: %w", err)
+		return
+	}
+
+	rc, err := client.Bucket(object.Bucket).Object(object.Name).NewReader(ctx)
+
+	if err != nil {
+		return fmt.Errorf("Object(%q).NewReader: %v", object.Name, err)
+	}
+	defer rc.Close()
+
+	if _, err := io.Copy(f, rc); err != nil {
+		return fmt.Errorf("io.Copy: %v", err)
+	}
+
+	if err = f.Close(); err != nil {
+		return fmt.Errorf("f.Close: %v", err)
+	}
+
+	file, err := json.MarshalIndent(*tempFileObjectMetadata, "", " ")
```

Comment:
The `file` is a variable that stores metadata, which has nothing to do with file. Prefer variable names like `metadata` or `file_content`